src/decoder/error.rs

```diff
@@ -1,39 +1,60 @@
 use std::{error, fmt, fmt::Display, io, string};
 
-use serde::de::{self, Expected, Unexpected};
+use crate::Bson;
+use de::Unexpected;
+use serde::de;
 
 /// Possible errors that can arise during decoding.
 #[derive(Debug)]
 #[non_exhaustive]
 pub enum DecoderError {
+    /// A `std::io::Error` encountered while deserializing.
```

Comment:
Should we make this (and the other std error types in the docs below) links?

Reply:
done